### PR TITLE
Juniper get_vlan(name) method

### PR DESCRIPTION
**Pull request by emmurd**

### CODE REVIEW

File: netman/adapters/switches/juniper/base.py [hunk 1]
```diff
@@ -113,6 +113,32 @@ def get_vlans(self):
 
         return vlan_list
 
+    def get_vlan(self, number):
+        config = self.query(one_vlan_by_vlan_id(number), all_interfaces)
+        if len(config.xpath("data/configuration/vlans/vlan")) >= 1:
+            vlan_node = config.xpath("data/configuration/vlans/vlan")[0]
+            number_node = first(vlan_node.xpath("vlan-id"))
```
Comment (lindycoder) [inline netman/adapters/switches/juniper/base.py]: You could probably have methos doing this and have get_vlans use it too... this seems like a lot of duplicate code

File: tests/adapters/switches/juniper_test.py [hunk 2]
```diff
@@ -263,6 +263,365 @@ def test_get_vlans_where_vlan_interfaces_can_also_be_called_irb(self):
...
+                      </family>
+                    </unit>
+                  </interface>
+                </interfaces>
+            """))
+
+        vlan = self.switch.get_vlan(20)
+
+        assert_that(vlan.number, equal_to(20))
+        assert_that(vlan.name, equal_to(None))
```
Comment (lindycoder) [inline tests/adapters/switches/juniper_test.py]: shouldn't the vlan name be "WITH-IF" ?

File: tests/adapters/switches/juniper_test.py [hunk 3]
```diff
@@ -263,6 +263,365 @@ def test_get_vlans_where_vlan_interfaces_can_also_be_called_irb(self):
...
+                </unit>
+              </interface>
+            </interfaces>
+        """))
+
+        vlan = self.switch.get_vlan(20)
+
+        assert_that(str(vlan.ips[0].ip), equal_to("2.1.1.1"))
+
+    def test_get_vlan_where_vlan_interfaces_can_also_be_called_irb_not_found(self):
```
Comment (lindycoder) [inline tests/adapters/switches/juniper_test.py]: name does not reflect test content, should not mention "irb" in the test name

File: tests/adapters/switches/juniper_test.py (hunk 2 above)
Comment (emmurd) [inline tests/adapters/switches/juniper_test.py]: The name that is return in the Vlan object is taken from the <description> tag sent from Netconf. In this case, no description is given, so there's no name in the Vlan object.

Comment (lindycoder) [inline tests/adapters/switches/juniper_test.py]: Ah correct

**Comment by emmurd** [comment]
New version with fixes and rebase

**Comment by lindycoder** [comment]
:+1:

**Comment by stephanerobert** [comment]
:+1:

**Pull request merged**